### PR TITLE
[Merged by Bors] - refactor: fold isStrictIntComparison into mkNonstrictIntProof in Linarith.Preprocessing

### PR DESCRIPTION
**Pull request by dwrensha**
Implements the refactor proposed in https://github.com/leanprover-community/mathlib4/pull/8310#issuecomment-1817330246.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Linarith/Preprocessing.lean [hunk 1]
```diff
@@ -192,51 +192,33 @@ end natToInt
...
         (← mkAppM ``lt_of_not_ge #[pf])
-    | (``GE.ge, #[_, _, a, b]) =>
+    | (``GE.ge, #[.const ``Int [], _, a, b]) =>
       return mkApp (← mkAppM ``Iff.mpr #[← mkAppOptM ``Int.add_one_le_iff #[a, b]])
         (← mkAppM ``lt_of_not_ge #[pf])
-    | _ => throwError "mkNonstrictIntProof failed: proof is not an inequality"
-  | _ => throwError "mkNonstrictIntProof failed: proof is not an inequality"
-
+    | _ => return pf
+  | _ => return pf
```
Comment (kmill) [inline Mathlib/Tactic/Linarith/Preprocessing.lean:215]: What you did looks correct in that last transformation, but I think we should back it out to right before that last commit. The `MetaM (Option Expr)` signature gives strictly more information to the caller, which could perhaps be useful one day.

What you can do in `strengthenStrictInt` to keep it similarly short is `return [(← mkNonstrictIntProof h).getD h]`

Comment (dwrensha) [reply Mathlib/Tactic/Linarith/Preprocessing.lean]: done

**Comment by kmill** [comment]
Great, thanks. Once CI passes, I think we can merge.

bors d+

**Pull request closed without merging**